planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/utils/calculation.cpp

```diff
@@ -391,18 +391,37 @@ double calc_lane_changing_acceleration(
     prepare_longitudinal_acc);
 }
 
+double calc_actual_prepare_duration(
+  const CommonDataPtr & common_data_ptr, const double current_velocity,
+  const double active_signal_duration)
+{
+  const auto max_prepare_duration = common_data_ptr->lc_param_ptr->maximum_prepare_duration;
+  const auto min_lc_velocity = common_data_ptr->lc_param_ptr->minimum_lane_changing_velocity;
+
+  // need to ensure min prep duration is sufficient to reach minimum lane changing velocity
+  const auto min_prepare_duration = std::invoke([&]() -> double {
+    if (current_velocity >= min_lc_velocity) {
+      return common_data_ptr->lc_param_ptr->minimum_prepare_duration;
+    }
+    const auto max_acc = common_data_ptr->bpp_param_ptr->max_acc;
+    return (min_lc_velocity - current_velocity) / max_acc;
```

Comment:
there mighttt be possibility `max_acc = 0.0`? should we place a guard or something?